src/Fixer/Operator/BinaryOperatorSpacesFixer.php

```diff
@@ -242,40 +248,36 @@ protected function applyFix(\SplFileInfo $file, Tokens $tokens)
      */
     protected function createConfigurationDefinition()
     {
-        $allowedValues = [
+        static $allowedValues = [
             self::SINGLE_SPACE,
             AlignmentFixerHelper::ALIGN,
             AlignmentFixerHelper::ALIGN_SINGLE_SPACE,
             AlignmentFixerHelper::ALIGN_SINGLE_SPACE_MINIMAL,
             null,
         ];
 
-        $options = [];
-        foreach (self::$defaultConfiguration as $option => $value) {
-            $options[] =
-                (new FixerOptionBuilder($option, $option))
-                    ->setDefault($value)
-                    ->setAllowedValues($allowedValues)
-                    ->getOption()
-                ;
-        }
-
-        // add deprecated options as BC layer
-        $options[] =
+        return new FixerConfigurationResolver([
+            (new FixerOptionBuilder('default', 'default fix strategy.'))
+                ->setDefault(self::SINGLE_SPACE)
+                ->setAllowedValues($allowedValues)
+                ->getOption(),
+            (new FixerOptionBuilder('operators', 'List of binary operator to fix.'))
+                ->setAllowedTypes(['array'])
+//                ->setAllowedValues([
+//                    (new FixerOptionValidatorGenerator())->allowedValueIsSubsetOf($a), FIXME
```

Comment:
fixme ?

Reply:
yeah, ping @julienfalque how to setup allowed values for associative array? couldn't find where we use something like this ATM

Reply:
the format will be like:
```
operators => {
    '=' => string|null,
    '*' => string|null,
    ...
}
```

right ?

Reply:
indeed, with the `string` value being one of 4 values

Reply:
At the moment there is no validator for options that are associative arrays. A solution could be to add a normalizer to validate it manually, maybe using a custom `OptionsResolver` in it?

Reply:
Thanks I give it a try! : )
TBH I do 't like `normalizers` to do validation as to have a clear separation of concerns. As such I think I'll do validation with the `configure` method.

Reply:
all updated